src/votekit/cvr_loaders.py

```diff
@@ -68,3 +69,89 @@ def rank_column_csv(
         ballots.append(b)
 
     return PreferenceProfile(ballots=ballots)
+
+
+def blt(fpath: str) -> tuple[PreferenceProfile, int]:
+    """
+    given a blt file path, loads cvr. (blt is text-like format used for scottish election data)
+    the first line of the file is metadata recording the number of candidates and seats,
+    followed by ballot data (first number in row is ballot weight),
+    followed by candidate data (order corresponds to number in ballots),
+    followed by election location
+    Args:
+        fpath (str): path to cvr file
+    Raises:
+        FileNotFoundError: if fpath is invalid
+        EmptyDataError: if dataset is empty
+        DataError: if there is missing or incorrect metadata or candidate data
+    Returns:
+        PreferenceProfile: a preference schedule that
+        represents all the ballots in the elction
+        seats: number of seats in the election
+    """
+    ballots = []
+    names = []
+    name_map = {}
+    numbers = True
+    cands_included = False
+
+    if not os.path.isfile(fpath):
+        raise FileNotFoundError(f"File with path {fpath} cannot be found")
+    if os.path.getsize(fpath) == 0:
+        raise EmptyDataError("Dataset cannot be empty")
+
+    with open(fpath, "r") as file:
+        for i, line in enumerate(file):
+            s = line.rstrip("\n").rstrip()
```

Comment:
why are we striping the line twice here?

Reply:
I believe the first rstrip call strips the newline character, and the second strips any whitespace from the end of the line. (Jamie wrote this part of the code though, not me)